GitUI/UserControls/RevisionGridClasses/FormQuickGitRefSelector.cs

```diff
@@ -0,0 +1,128 @@
+﻿using System;
+using System.Drawing;
+using System.Linq;
+using System.Windows.Forms;
+using GitUIPluginInterfaces;
+using ResourceManager;
+
+namespace GitUI.UserControls.RevisionGridClasses
+{
+    public partial class FormQuickGitRefSelector : GitModuleForm
+    {
+        private readonly TranslationString _actionRename = new TranslationString("Rename");
+        private readonly TranslationString _actionDelete = new TranslationString("Delete");
+        private readonly TranslationString _tag = new TranslationString("tag");
+
+
+        public FormQuickGitRefSelector()
+        {
+            InitializeComponent();
+            Translate();
+        }
+
+
+        /// <summary>
+        /// Gets the ref selected by the user.
+        /// </summary>
+        public IGitRef SelectedRef => (lbxRefs.SelectedItem as DisplyGitRef)?.Item;
+
+
+        public void Init(Action action, IGitRef[] refs)
+        {
+            switch (action)
+            {
+                case Action.Delete:
+                    {
+                        btnAction.Text = _actionDelete.Text;
+                    }
+                    break;
+                case Action.Rename:
+                    {
+                        btnAction.Text = _actionRename.Text;
+                    }
+                    break;
+            }
+
+            lbxRefs.Items.Clear();
+            if (refs == null || refs.Length < 1)
+            {
+                DialogResult = DialogResult.Cancel;
+                Close();
+                return;
+            }
+
+            // constrain the listbox min width, otherwise the button may not have enough space
+            var longestItemWidth = 150f;
+            try
+            {
+                lbxRefs.BeginUpdate();
+
+                using (Graphics graphics = lbxRefs.CreateGraphics())
+                {
+                    foreach (var gitRef in refs.OrderBy(r => r.IsTag).ThenBy(r => r.Name))
+                    {
+                        var suffix = gitRef.IsTag ? $" ({_tag.Text})" : string.Empty;
+                        var item = new DisplyGitRef
+                        {
+                            Label = $"{gitRef.Name}{suffix}",
+                            Item = gitRef
+                        };
+                        lbxRefs.Items.Add(item);
+
+                        // assume that the branch names or tags are never longer than 100 symbols long
+                        // if they are (sanity!) - don't resize past 100 symbols
+                        var label = item.Label.Length > 100 ? item.Label.Substring(0, 100) : item.Label;
```

Comment:
I think we're going somewhat OT here, but I'll take the bait :)

>What the chances are that other developer will search for such a piece of code?

From 0% all the way to 100% :) 
First and foremost, it will likely depend on how well one knows the codebase. 
In a large unfamiliar codebase, unless intellisence puts a method right in front of my eyes or I stumble across a method which may be useful - unlikely. 
Secondly, if I can relatively easy implement the required - I will do so. If not - I may search. Probably SO first, before the codebase :)
Extensions would only come up in intellisence if the required namespace is already referenced, or by some slim chance I happened to type the exact extension method name and I get prompted by Re# to import the namespace...

It is my firm belief that a codebase must have a dedicated owner/s who set guidelines. The owners must be judgemental - it is them who own and shape the code. This way the code is likely to be maintainable and consistent.
And, as a drive-by contributor, I expect guidance from the codebase guardians to tap me on a shoulder and say - "hey, we already have a class/method/extension which provides the necessary implementation" or "this can be done in such and such way....". 
Without a strong ownership the codebase risks becoming bloated.

I speak from experience maintaining a large enterprise \*cough\*framework\*cough\*.